src/Webinex.Asky/Factories/FilterExpressions.cs

```diff
@@ -198,4 +215,41 @@ public static Expression<Func<TEntity, bool>> All<TEntity>(
         var methodCallExpression = Expression.Call(null, anyMethod, new[] { propertyAccessExpression, predicate });
         return Expression.Lambda<Func<TEntity, bool>>(methodCallExpression, parameter);
     }
+
+    /// <summary>
+    /// It make sense to wrap value to container instead of using Expression.Constant directly, because in this case,
+    /// tools like Entity Framework later will be able to optimize it.
+    /// </summary>
+    private static MemberExpression WrapValueToContainerMember(object value, Type type)
+    {
+        var accessor = ValueContainerFactoriesCache.GetOrAdd(type, (Type t) =>
+        {
+            var param = Expression.Parameter(typeof(object));
+            var ctorInfo = typeof(ValueContainer<>).MakeGenericType(t)
+                .GetConstructor(new[] { t }) ?? throw new ArgumentNullException();
+
+            var expression = Expression.Lambda<Func<object, object>>(
+                Expression.New(ctorInfo, new[] { Expression.Convert(param, t) }),
+                new[] { param });
+            return expression.Compile();
+        });
+
+        var containerInstance = accessor(value);
+        return Expression.MakeMemberAccess(
+            Expression.Constant(containerInstance),
+            containerInstance.GetType().GetMember(nameof(ValueContainer<object>.Value)).First());
```

Comment:
containerInstance.GetType().GetMember(nameof(ValueContainer<object>.Value)) can be cached as well (if needed)